modin/backends/pandas/query_compiler.py

```diff
@@ -1097,21 +1097,25 @@ def unstack(self, level, fill_value):
         def map_func(df):
             return pandas.DataFrame(df.unstack(level=level, fill_value=fill_value))
 
-        is_correct_len = True
-        if isinstance(self.index, pandas.MultiIndex):
+        def is_tree_like(calc_index, valid_index):
+            if not isinstance(calc_index, pandas.MultiIndex):
```

Comment:
Why if `calc_index` is not `MultiIndex`, index will be tree-like? I think an index, which is `MultiIndex`, may be either tree-like or not tree-like, is not it so?

Reply:
`is_tree_like` is needed for behavior determination for tree-like `MultiIndex` and simple Index. At all, I might change `is_tree_like`->`is_tree_like_or_index`

Reply:
May be just change to `is_not_tree_like`?

Reply:
Or `is_tree_like_or_not_mi`

Reply:
Names of variables and function was changed in style `is_`axis`_tree_like_or_idx`.